source/RevitLookup/Core/ComponentModel/Descriptors/DatumPlaneDescriptor.cs

```diff
@@ -0,0 +1,174 @@
+{
+    public ResolveSet Resolve(Document context, string target, ParameterInfo[] parameters)
+    {
+        return target switch
+        {
+            nameof(DatumPlane.CanBeVisibleInView) => ResolveCanBeVisibleInView(),
+            nameof(DatumPlane.GetCurvesInView) => ResolveCurvesInView(),
+            nameof(DatumPlane.GetDatumExtentTypeInView) => ResolveDatumExtentTypeInView(),
+            nameof(DatumPlane.GetLeader) => ResolveLeader(),
+            nameof(DatumPlane.GetPropagationViews) => ResolvePropagationViews(),
+            nameof(DatumPlane.HasBubbleInView) => ResolveHasBubbleInView(),
+            nameof(DatumPlane.IsBubbleVisibleInView) => ResolveBubbleVisibleInView(),
+            _ => null
+        };
+        
+        ResolveSet ResolveCanBeVisibleInView()
+        {
+            var views = context.EnumerateInstances<View>().ToArray();
+            var resolveSummary = new ResolveSet(views.Length);
+            
+            foreach (var view in views)
+            {
+                var result = datumPlane.CanBeVisibleInView(view);
+                resolveSummary.AppendVariant(result, $"{view.Name}: {result}");
+            }
+            
+            return resolveSummary;
+        }
+        
+        ResolveSet ResolveDatumExtentTypeInView()
+        {
+            var resolveSummary = new ResolveSet(2);
+            
+            if (datumPlane.CanBeVisibleInView(context.ActiveView))
+            {
+                var resultEnd0 = datumPlane.GetDatumExtentTypeInView(DatumEnds.End0, context.ActiveView);
+                var resultEnd1 = datumPlane.GetDatumExtentTypeInView(DatumEnds.End1, context.ActiveView);
+                resolveSummary.AppendVariant(resultEnd0, $"End 0, Active view: {resultEnd0}");
+                resolveSummary.AppendVariant(resultEnd1, $"End 1, Active view: {resultEnd1}");
+            }
+            else
+            {
+                resolveSummary.AppendVariant("Datum plane can not be visible on active view");
+            }
+            
+            return resolveSummary;
+        }
+        
+        ResolveSet ResolveCurvesInView()
+        {
+            var resolveSummary = new ResolveSet(2);
+            
+            if (datumPlane.CanBeVisibleInView(context.ActiveView))
+            {
+                resolveSummary.AppendVariant(datumPlane.GetCurvesInView(DatumExtentType.Model, context.ActiveView), $"Model, Active view");
+                resolveSummary.AppendVariant(datumPlane.GetCurvesInView(DatumExtentType.ViewSpecific, context.ActiveView), $"ViewSpecific, Active view");
+            }
+            else
+            {
+                resolveSummary.AppendVariant("Datum plane can not be visible on active view");
+            }
+            
+            return resolveSummary;
+        }
+        
+        ResolveSet ResolveLeader()
+        {
+            if (datumPlane is ReferencePlane)
```

Comment:
Why you check a specific type?

Reply:
I check it because Reference plane throws an exception when I try to use these methods on it. I just copied text of exception